float32 x
float32 y
float32 dist
float32 relative_theta